logicalplan/plan.go

```diff
@@ -275,6 +278,60 @@ func trimSorts(expr Node) Node {
 	return expr
 }
 
+func reduceConstantExpressions(expr Node) Node {
+	TraverseBottomUp(nil, &expr, func(parent, current *Node) bool {
+		if current == nil || parent == nil {
+			return true
+		}
+		switch tparent := (*parent).(type) {
+		case *Parens:
+			num, err := UnwrapFloat(tparent.Expr)
+			if err != nil {
+				return false
+			}
+			*parent = &NumberLiteral{Val: num}
+		case *Unary:
+			num, err := UnwrapFloat(tparent.Expr)
+			if err != nil {
+				return false
+			}
+			switch tparent.Op {
+			case parser.ADD:
+				*parent = &NumberLiteral{Val: num}
+			case parser.SUB:
+				*parent = &NumberLiteral{Val: -num}
+			}
+		case *Binary:
+			lnum, err := UnwrapFloat(tparent.LHS)
+			if err != nil {
+				return false
+			}
+			rnum, err := UnwrapFloat(tparent.RHS)
+			if err != nil {
+				return false
+			}
+			switch tparent.Op {
+			case parser.ADD:
+				*parent = &NumberLiteral{Val: lnum + rnum}
```

Comment:
Could this impact accuracy if there are multiple additions and we don't use something like Kahan summation?

Reply:
This is what binary expression would evaluate too at query time too. Since we only likely can have a few summands here it's probably fine!